bugreportutils/bugreportutils.go

```diff
@@ -43,35 +43,51 @@ var (
 	// BugReportSectionRE is a regular expression to match the beginning of a bug report section.
 	BugReportSectionRE = regexp.MustCompile(`------\s+(?P<section>.*)\s+-----`)
 
-	// deviceIDRE is a regular expression that matches the "DeviceID" line
+	// deviceIDRE is a regular expression that matches the "DeviceID" line.
 	deviceIDRE = regexp.MustCompile("DeviceID: (?P<deviceID>[0-9]+)")
 
-	// sdkVersionRE is a regular expression that finds sdk version in the System Properties section of a bug report
+	// sdkVersionRE is a regular expression that finds sdk version in the System Properties section
+	// of a bug report.
 	sdkVersionRE = regexp.MustCompile(`\[ro.build.version.sdk\]:\s+\[(?P<sdkVersion>\d+)\]`)
 
-	// buildFingerprintRE is a regular expression to match any build fingerprint line in the bugreport
+	// buildFingerprintRE is a regular expression to match any build fingerprint line in the bugreport.
 	buildFingerprintRE = regexp.MustCompile(`Build\s+fingerprint:\s+'(?P<build>\S+)'`)
 
-	// modelNameRE is a regular expression that finds the model name line in the System Properties section of a bug report.
+	// modelNameRE is a regular expression that finds the model name line in the System Properties
+	// section of a bug report.
 	modelNameRE = regexp.MustCompile(`\[ro.product.model\]:\s+\[(?P<modelName>.*)\]`)
 
 	// pidRE is a regular expression to match PID to app name and UID.
-	pidRE = regexp.MustCompile(`PID #` + `(?P<pid>\d+)` + `: ProcessRecord[^:]+:` + `(?P<app>[^/]+)` + `/` + `(?P<uid>.*)` + `}`)
+	pidRE = regexp.MustCompile(`PID #` + `(?P<pid>\d+)` + `: ProcessRecord[^:]+:` +
+		`(?P<app>[^/]+)` + `/` + `(?P<uid>.*)` + `}`)
 
-	// sensorLineMMinusRE is a regular expression to match the sensor list line in the sensorservice dump of a bug report from MNC or before.
-	sensorLineMMinusRE = regexp.MustCompile(`(?P<sensorName>[^|]+)` + `\|` + `(?P<sensorManufacturer>[^|]+)` + `\|` +
-		`(\s*version=(?P<versionNumber>\d+)\s*\|)?` + `\s*(?P<sensorTypeString>[^|]+)` +
-		`\|` + `\s*(?P<sensorNumber>0x[0-9A-Fa-f]+)`)
-
-	// sensorLineNPlusRE is a regular expression to match the sensor list line in the sensorservice dump of a bug report starting from NRD42 and onwards.
-	sensorLineNPlusRE = regexp.MustCompile(`(?P<sensorNumber>0x?[0-9A-Fa-f]+)` + `\)\s*` +
+	// sensorLineOneRE is a regular expression to match the first line of sensor information
+	// from the sensor list line in the sensorservice dump in the bugreport starting
+	// from NRD42 and onwards.
+	sensorLineOneRE = regexp.MustCompile(`(?P<sensorHandle>0x?[0-9A-Fa-f]+)` + `\)\s*` +
```

Comment:
Why change sensorNumber to sensorHandle?

ALso deleting sensorLineMMinus will break backward compatibility.

Reply:
I was use both "handle" and "number" to refer to the unique identification number for each sensor. To keep things consistent, I will only be using the word "number" from now on.

Regarding the backward compatibility, I will need to look at more examples for older bug report to determine how to parse the information.